### PR TITLE
[SPARK-9245] [MLLIB] LDA topic assignments

### PR DESCRIPTION
**Pull request by jkbradley**
For each (document, term) pair, return top topic.  Note that instances of (doc, term) pairs within a document (a.k.a. "tokens") are exchangeable, so we should provide an estimate per document-term, rather than per token.

CC: @rotationsymmetry @mengxr

### CODE REVIEW

**Comment by jkbradley** [comment]
did not actually fail

**Comment by jkbradley** [comment]
test this please

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala [hunk 1]
```diff
@@ -575,6 +575,48 @@ class DistributedLDAModel private[clustering] (
...
+        val term: Int = index2term(edgeContext.dstId)
+        edgeContext.sendToSrc((Array(term), Array(topTopic)))
+      }
+    val mergeMsg: ((Array[Int], Array[Int]), (Array[Int], Array[Int])) => (Array[Int], Array[Int]) =
+      (terms_topics0, terms_topics1) => {
+        (terms_topics0._1 ++ terms_topics1._1, terms_topics0._2 ++ terms_topics1._2)
+      }
+    // M-STEP: Aggregation computes new N_{kj}, N_{wk} counts.
+    graph.aggregateMessages[(Array[Int], Array[Int])](sendMsg, mergeMsg).filter(isDocumentVertex)
+        .map { case (docID: Long, (terms: Array[Int], topics: Array[Int])) =>
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: fix indentation

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala [hunk 2]
```diff
@@ -575,6 +575,48 @@ class DistributedLDAModel private[clustering] (
...
+      val (sortedTerms, sortedTopics) = terms.zip(topics).sortBy(_._1).unzip
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: Leave a TODO because `zip` is not efficient

**Comment by mengxr** [comment]
LGTM except two minor comments.

**Comment by jkbradley** [comment]
OK, updated!  I created a temp val for the first comment since, otherwise, it was hard to get the indentation to behave nicely.

**Comment by rotationsymmetry** [comment]
LGTM.

**Comment by mengxr** [comment]
Merged into master and branch-1.5. Thanks!

**Pull request closed without merging**